src/firefly/java/edu/caltech/ipac/firefly/server/util/DownloadScript.java

```diff
@@ -161,6 +163,10 @@ static UrlInfo resolve(FileInfo fi, File baseDir) {
         String baseDirPath = ifNotNull(baseDir).get(f -> f.getPath().endsWith("/") ? f.getPath() : f.getPath() + "/");
         String href = ifNotNull(fi.getInternalFilename()).get(ServerContext::resolveUrl);
         String extName = ifNotNull(fi.getExternalName()).getOrElse(baseDirPath);
+        String suffix = fi.getSuffix();
+        if (!StringUtils.isEmpty(suffix)) {
+            extName = URLDownload.appendSuffixBeforeExtension(extName, suffix);
+        }
```

Comment:
This will not work consistently. Logic needs to be moved to passing suffix in to the download script, as discussed earlier. I'm working on changing this.